app/javascript/src/AccessTokens/components/ExpirationDatePicker.tsx

```diff
@@ -0,0 +1,176 @@
+import { useState, useMemo } from 'react'
+import { Alert, CalendarMonth, FormGroup, FormSelect, FormSelectOption, Popover } from '@patternfly/react-core'
+import OutlinedQuestionCircleIcon from '@patternfly/react-icons/dist/js/icons/outlined-question-circle-icon'
+
+import { createReactWrapper } from 'utilities/createReactWrapper'
+
+import type { FunctionComponent, FormEvent } from 'react'
+
+import './ExpirationDatePicker.scss'
+
+interface ExpirationItem {
+  id: string;
+  label: string;
+  period: number; // In seconds
+}
+
+const collection: ExpirationItem[] = [
+  { id: '7', label: '7 days', period: 7 },
+  { id: '30', label: '30 days', period: 30 },
+  { id: '60', label: '60 days', period: 60 },
+  { id: '90', label: '90 days', period: 90 },
+  { id: 'custom', label: 'Custom...', period: 0 },
+  { id: 'no-exp', label: 'No expiration', period: 0 }
+]
+
+const dayMs = 60 * 60 * 24 * 1000
+
+interface Props {
+  id: string;
+  label: string | null;
+  tzOffset?: number;
+}
+
+const ExpirationDatePicker: FunctionComponent<Props> = ({ id, label, tzOffset }) => {
+  const today: Date = new Date()
+  const tomorrow: Date = new Date(today)
+  tomorrow.setDate(today.getDate() + 1)
```

Comment:
This is be redefined every time the component re-renders, do we want this? Otherwise I would suggest creating a state.
```
const [today, _] = useState(new Date())
```

Reply:
In which event does the component re-render? only when the screen is loaded? or once after every state change?

Reply:
https://github.com/3scale/porta/pull/3943/commits/cca221d51a8f4505d80939e2cb0ea9c4a48e64ff

Reply:
> In which event does the component re-render? only when the screen is loaded? or once after every state change?

With every state change.

Reply:
Thanks, I didn't know this and that's crucial to understand how the whole thing works.